### PR TITLE
implement deletion of dns entries and REST api for it

### PR DESCRIPTION
**Pull request by ablepharus**

### CODE REVIEW

**Comment by frankosterfeld** [comment]
Context: this is needed to unregister signals in the opendigitizer service, when the signal metadata of a sink changes, or when the whole flow graph is replaced.

File: src/services/include/services/dns_client.hpp [hunk 1]
```diff
@@ -90,12 +112,21 @@ struct DnsClient {
...
+    std::vector<Entry> deleteSignals(const std::vector<Entry> &entries, std::chrono::seconds timeout = 20s) {
+        return _doSync<std::vector<Entry>>(&DnsClient::deleteSignalsAsync, entries, timeout);
+    }
     Entry registerSignal(const Entry &entry) {
         auto s = registerSignals({ entry });
         if (s.size() > 0)
             return s[0];
         return {};
     }
+    Entry deleteSignal(const Entry &entry) {
```
Comment (frankosterfeld) [inline src/services/include/services/dns_client.hpp:124]: Thinking about it, I'd prefer `unregisterSignal(s)`. It's symmetric with registerSignal(s), and also sounds less threatening :)

**Comment by frankosterfeld** [comment]
besides unregister vs. delete wording preference, looks good to me

File: src/services/include/services/dns_storage.hpp [hunk 2]
```diff
@@ -60,6 +60,30 @@ class DataStorage {
...
+        });
+        return deletedEntries;
+    }
+
+    template<class ReturnEntryType = EntryType, class FilterT = FilterType>
+    std::vector<ReturnEntryType> deleteEntries(const FilterT &filter = {}) {
+        std::vector<ReturnEntryType> deletedEntries;
+        std::erase_if(_entries,
+                [&filter, &deletedEntries](const StorageEntryType &entry) {
+                    if (filter == entry) {
```
Comment (wirew0rm) [inline src/services/include/services/dns_storage.hpp:78]: Do we really want to check for full equality here and not just delete by name?

Comment (ablepharus) [reply src/services/include/services/dns_storage.hpp:78]: I could change the default type to QueryEntry, that way you could delete by any field, or pass the exact entry.

**Review by wirew0rm — APPROVED** [review]

**Pull request merged**